### DISCUSSION TITLE
Issue building PuTTY on Mac.

### DISCUSSION THREAD

**Discussion by abcbarryn** [Homebrew/discussions#2528] [Everyday usage]
Since the issue/bug reporting page is not working (refer to my other post) I am posting this here. Currently when you run "brew install putty" you get several putty command line utilities but no actual PuTTY or Pterm. I have actually gotten this to WORK though  by installing  gtk-mac-bundler and editting the putty config/make scripts. I would like to suggest that a formulae be added for gtk-mac-bundler and the PuTTY formulae be updated to build the complete software package including the GUI client.
<img width="591" alt="Screen Shot 2021-11-24 at 9 34 23 AM" src="https://user-images.githubusercontent.com/13906545/143257888-e4b4bc14-1f8d-44df-a40a-2408fb4c460a.png">
<img width="1137" alt="Screen Shot 2021-11-24 at 9 35 38 AM" src="https://user-images.githubusercontent.com/13906545/143258305-9b66d078-3030-43a0-9884-040d7bef0f3f.png">

**Comment by carlocab** [comment]
If `gtk-mac-bundler` satisfies the requirements in https://docs.brew.sh/Acceptable-Formulae then we'd be happy to review a pull request that adds the formula to Homebrew/core.

As for the GUI, we usually avoid building those on purpose: https://docs.brew.sh/Acceptable-Formulae#stuff-that-builds-a-gui-by-default-but-doesnt-have-to That said, my impression is that PuTTY is more widely known for its GUI, so a PR that adds the GUI build to the formula might have a chance of being accepted. If there is an official pre-built binary for PuTTY we might want to make that a cask instead though.

**Reply by abcbarryn** [reply]
Well, I don't know about an "official" Mac binary, but I do have a 64 bit intel one that works and should run as a stand alone app or in conjunction with the current brew formulae. How do I submit it? The gtk-mac-bundler did get the PuTTY GUI to build after I did some edits, pulled out some hair, did a few more edits, wrote a custom loader script, and sacrificed a rubber chicken. :) PuTTY is certainly mainly useful for it's GUI.

**Reply by abcbarryn** [reply]
Here's a link to my dmg image I created.

[Mac PuTTY DMG](https://drive.google.com/file/d/1-4JzhRyw13T9OySpd8ExdxINM0pZqwOl/view?usp=sharing)

**Reply by syoung1999** [reply]
Thanks for providing the Putty DMG file, it worked perfectly.  I too was at the same point as you, I did the "brew install putty" and there was no putty command to be found.  Thanks again.
